diff -rupN navit/navit/graphics.c navit.opentom/navit/graphics.c
--- navit/navit/graphics.c	2014-03-11 11:48:44.000000000 +0100
+++ navit.opentom/navit/graphics.c	2014-03-18 14:53:45.000000000 +0100
@@ -612,19 +612,22 @@ image_new_helper(struct graphics *gra, s
 	for (i = 1 ; i <= 6 ; i++) {
 		char *new_name=NULL;
 		switch (i) {
-			case 1:
+			case 2:
 				/* The best variant both for cpu usage and quality would be prescaled png of a needed size */
 				if (width != -1 && height != -1) {
 					new_name=g_strdup_printf("%s_%d_%d.png", name, width, height);
-				}
+				} else {
+                                        new_name=g_strdup_printf("%s_%d_%d.png", name, 16, 16);
+                                }
 				break;
-			case 2:
+			case 1:
 				/* Try to load image by the exact name given by user. For example, if she wants to
 				  scale some prescaled png variant to a new size given as function params, or have
 				  default png image to be displayed unscaled. */
 				new_name=g_strdup(path);
 				break;
 			case 3:
+#ifdef HAVE_QT_SVG
 				/* Next, try uncompressed and compressed svgs as they should give best quality but 
 				   rendering might take more cpu resources when the image is displayed for the first time */
 				new_name=g_strdup_printf("%s.svg", name);
@@ -632,6 +635,9 @@ image_new_helper(struct graphics *gra, s
 			case 4:
 				new_name=g_strdup_printf("%s.svgz", name);
 				break;
+#else
+                                i = 5;
+#endif
 			case 5:
 				/* Scaling the default png to the needed size may give some quality loss */
 				new_name=g_strdup_printf("%s.png", name);
