diff --git a/navit/CMakeLists.txt b/navit/CMakeLists.txt
index ca19ce41..13276770 100644
--- a/navit/CMakeLists.txt
+++ b/navit/CMakeLists.txt
@@ -107,7 +107,7 @@ if(NOT MSVC)
 	SET(NAVIT_LIBS ${NAVIT_LIBS} m)
 endif(NOT MSVC)
 target_link_libraries(${NAVIT_LIBNAME}  ${MODULES_NAME} ${NAVIT_SUPPORT_LIBS} fib ${NAVIT_LIBS} )
-set_target_properties(${NAVIT_LIBNAME} PROPERTIES COMPILE_DEFINITIONS "MODULE=navit;LIBDIR=\"${CMAKE_INSTALL_PREFIX}/${LIB_DIR}\";PREFIX=\"${CMAKE_INSTALL_PREFIX}\"")
+set_target_properties(${NAVIT_LIBNAME} PROPERTIES COMPILE_DEFINITIONS "MODULE=navit;LIBDIR=\"/mnt/sdcard/opentom/lib\";PREFIX=\"/mnt/sdcard/opentom\"")
 if (DEFINED NAVIT_COMPILE_FLAGS)
 	set_target_properties(${NAVIT_LIBNAME} PROPERTIES COMPILE_FLAGS "${NAVIT_COMPILE_FLAGS}")
 endif()

--- ./navit/xslt/tomtom.xslt.orig	2020-08-21 23:09:21.646678108 +0000
+++ ./navit/xslt/tomtom.xslt	2020-08-21 23:14:14.994671983 +0000
@@ -11,11 +11,12 @@
 
 	<xsl:template match="/">
 		<xsl:apply-templates select="config"/>
-		<xsl:apply-templates select="layout"/>
+		<xsl:copy-of select="layout"/>
 	</xsl:template>
 
 	<xsl:template match="config">
 		<xsl:copy>
+			<xsl:copy-of select="layout"/>
 			<xsl:apply-templates select="navit"/>
 		</xsl:copy>
 	</xsl:template>
@@ -27,11 +28,11 @@
 			<xsl:attribute name="orientation">-1</xsl:attribute>
 			<xsl:attribute name="autozoom_active">1</xsl:attribute>
 			<xsl:attribute name="recent_dest">25</xsl:attribute>
-			<graphics type="sdl" w="480" h="272" bpp="16" frame="0" flags="1"/>
-			<xsl:copy-of select="gui[@type='internal']"/>
+			<graphics type="sdl" w="320" h="240" bpp="16" frame="0" flags="1"/>
+			<xsl:apply-templates select="gui[@type='internal']"/>
 			<vehicle name="Local GPS" profilename="car" enabled="yes" active="yes" follow="1" source="file:/var/run/gpspipe">
 			<!-- Navit can write a tracklog in several formats (gpx, nmea or textfile): -->
-				<log enabled="no" type="gpx" attr_types="position_time_iso8601,position_direction,position_speed,position_radius" data="/mnt/sdcard/navit/track_%Y%m%d-%%i.gpx" flush_size="1000" flush_time="30"/>
+				<log enabled="no" type="gpx" attr_types="position_time_iso8601,position_direction,position_speed,position_radius" data="$NAVIT_PREFIX/track_%Y%m%d-%%i.gpx" flush_size="1000" flush_time="30"/>
 			</vehicle>
 			<vehicle name="Demo" profilename="car" enabled="yes" active="no" follow="1" source="demo://" speed="100"/>
 			<xsl:copy-of select="tracking"/>
@@ -46,20 +47,24 @@
 			<xsl:copy-of select="navigation"/>
 
 			<xsl:comment>Use espeak.</xsl:comment>
-			<speech type="cmdline" data="/mnt/sdcard/navit/bin/espeakdsp -v en '%s'"/>
+			<speech type="cmdline" data="$NAVIT_PREFIX/bin/espeakdsp -v en '%s'"/>
 
 			<mapset enabled="yes">
 				<map type="binfile" enabled="yes" data="$NAVIT_SHAREDIR/maps/*.bin"/>
 			</mapset>
 			<xsl:copy-of select="layer"/>
+			<xi:include href="$NAVIT_SHAREDIR/navit_layout_*.xml"/>
 			<xsl:copy-of select="layout"/>
 			<xsl:copy-of select="xi:include"/>
 		</xsl:copy>
 	</xsl:template>
 
-	<xsl:template match="layout">
+	<xsl:template match="gui[@type='internal']">
 		<xsl:copy>
-			<xsl:copy-of select="@*|node()"/>
+			<xsl:copy-of select="@*"/>
+			<xsl:attribute name="fullscreen">1</xsl:attribute>
+			<xsl:attribute name="pitch">35</xsl:attribute>
+			<xsl:apply-templates/>
 		</xsl:copy>
 	</xsl:template>
 </xsl:transform>
