diff -rupN OpenDUNE.original/src/audio/driver.c OpenDUNE/src/audio/driver.c
--- OpenDUNE.original/src/audio/driver.c	2014-02-17 14:39:44.000000000 +0100
+++ OpenDUNE/src/audio/driver.c	2014-02-05 09:03:35.000000000 +0100
@@ -125,6 +125,7 @@ static bool Drivers_SoundMusic_Init(bool
 	g_bufferMusic->buffer = calloc(1, size);
 	g_bufferMusic->index  = 0xFFFF;
 
+        return false; /* CLM */
 	return true;
 }
 
@@ -142,6 +143,7 @@ static bool Drivers_Voice_Init(bool enab
 
 	if (!Drivers_Init(voice, "VOC")) return false;
 
+        return false; /* CLM */
 	return true;
 }
 
diff -rupN OpenDUNE.original/src/gui/security.c OpenDUNE/src/gui/security.c
--- OpenDUNE.original/src/gui/security.c	2014-02-17 14:39:44.000000000 +0100
+++ OpenDUNE/src/gui/security.c	2014-02-05 09:03:35.000000000 +0100
@@ -79,6 +79,7 @@ static void GUI_Security_NormaliseText(c
  */
 bool GUI_Security_Show(void)
 {
+#ifdef WITH_SECURITY
 	const char *wsaHouseFilename;
 	uint16 questionsCount;
 	uint16 oldCurrentWidget;
@@ -208,4 +209,7 @@ bool GUI_Security_Show(void)
 	g_interrogation = false;
 
 	return valid;
+#else
+        return true;
+#endif
 }
diff -rupN OpenDUNE.original/src/opendune.c OpenDUNE/src/opendune.c
--- OpenDUNE.original/src/opendune.c	2014-02-17 14:39:44.000000000 +0100
+++ OpenDUNE/src/opendune.c	2014-02-05 09:26:03.000000000 +0100
@@ -60,7 +60,7 @@
 #include "video/video.h"
 
 
-const char *window_caption = "OpenDUNE - Pre v0.8";
+const char *window_caption = "OpenDUNE - Pre v0.8 (OpenTom)";
 
 bool g_dune2_enhanced = true; /*!< If false, the game acts exactly like the original Dune2, including bugs. */
 
@@ -1121,10 +1121,12 @@ static void GameLoop_Main(void)
 
 static bool Unknown_25C4_000E(void)
 {
-	Timer_Init();
+	
 
 	if (!Video_Init()) return false;
 
+        Timer_Init();
+        
 	Mouse_Init();
 
 	/* Add the general tickers */
diff -rupN OpenDUNE.original/src/video/video_sdl.c OpenDUNE/src/video/video_sdl.c
--- OpenDUNE.original/src/video/video_sdl.c	2014-02-17 14:39:44.000000000 +0100
+++ OpenDUNE/src/video/video_sdl.c	2014-02-05 09:30:19.000000000 +0100
@@ -16,7 +16,8 @@
 #define SCREEN_USE_SCALE2X
 #undef SCREEN_USE_SCALE2X
 /** The the magnification of the screen. 2 means 640x400, 3 means 960x600, etc. */
-#define SCREEN_MAGNIFICATION 2
+/* CLM #define SCREEN_MAGNIFICATION 2 */
+#define SCREEN_MAGNIFICATION 1
 
 static SDL_Surface *s_gfx_surface = NULL;
 static uint8 *s_gfx_screen = NULL;
@@ -161,13 +162,13 @@ bool Video_Init(void)
 		return false;
 	}
 
-	SDL_WM_SetCaption(window_caption, "");
-	s_gfx_surface = SDL_SetVideoMode(SCREEN_WIDTH * SCREEN_MAGNIFICATION, SCREEN_HEIGHT * SCREEN_MAGNIFICATION, 8, SDL_SWSURFACE | SDL_HWPALETTE);
+	s_gfx_surface = SDL_SetVideoMode(SCREEN_WIDTH * SCREEN_MAGNIFICATION, SCREEN_HEIGHT * SCREEN_MAGNIFICATION, 8, SDL_SWSURFACE | SDL_HWPALETTE );
 	if (s_gfx_surface == NULL) {
 		Error("Could not set resolution: %s\n", SDL_GetError());
 		return false;
 	}
 
+	SDL_WM_SetCaption(window_caption, "");
 	SDL_ShowCursor(SDL_DISABLE);
 	SDL_EnableKeyRepeat(SDL_DEFAULT_REPEAT_DELAY, SDL_DEFAULT_REPEAT_INTERVAL);
 
@@ -436,6 +437,15 @@ static void Video_DrawScreen(void)
 		gfx1 = gfx3;
 	}
 }
+/* CLM BEGIN */
+#	elif SCREEN_MAGNIFICATION == 1 
+static void Video_DrawScreen(void)
+{
+    uint8 *data = GFX_Screen_Get_ByIndex(SCREEN_0);
+    uint8 *gfx  = s_gfx_screen;
+    memcpy(gfx, data, SCREEN_WIDTH*SCREEN_HEIGHT);
+}
+/* CLM END */
 #	else /* SCREEN_MAGNIFICATION != 2 != 3 */
 static void Video_DrawScreen(void)
 {
