diff -rupN csrinit/main.c csrinit.opentom/main.c
--- csrinit/main.c	2012-07-11 16:20:31.000000000 +0200
+++ csrinit.opentom/main.c	2014-02-27 14:48:22.000000000 +0100
@@ -1141,7 +1141,7 @@ void send_next_packet(struct ubcsp_packe
         case 24:
             /* set default transmit power in dBm */
             memset(data, 0x00, sizeof(data));
-            data[0] = 20; /* 20 dBm */
+            data[0] = 0; // CLM data[0] = 20; /* 20 dBm */
             csr_setps(send_packet, CSR_PS_LC_DEFAULT_TX_POWER,
                 1, data);
 #ifdef SUPPORT_USB
