--- nxlib/Text16.c	2014-02-15 07:58:00.000000000 +0100
+++ nxlib.opentom/Text16.c	2014-03-03 07:53:15.000000000 +0100
@@ -4,7 +4,7 @@ int
 XDrawString16(Display *dpy, Drawable d, GC gc, int x, int y,
 	_Xconst XChar2b *string, int length)
 {   
-printf("XDrawString16 %d %x %x\n", length, string->byte1, string->byte2);
+	DPRINTF("XDrawString16 %d %x %x\n", length, string->byte1, string->byte2);
 	if (length > 0)
 		GrText(d, gc->gid, x, y, (void *)string, length, 
 		       GR_TFXCHAR2B|GR_TFBASELINE);
--- nxlib/QueryFont.c	2014-01-07 16:45:59.000000000 +0100
+++ nxlib.ok/QueryFont.c	2014-02-26 11:23:30.000000000 +0100
@@ -12,12 +12,13 @@ XFontStruct *
 XQueryFont(Display * dpy, XID font_ID)
 {
 	int i, size;
-	XFontStruct *fs = (XFontStruct *) Xcalloc(1, sizeof(XFontStruct));
 	XCharStruct *cs;
 	GR_FONT_INFO finfo;
+	XFontStruct *fs;
 
 	GrGetFontInfo(font_ID, &finfo);
 
+	fs = (XFontStruct *) Xcalloc(1, sizeof(XFontStruct));
 	fs->fid = font_ID;
 
 	/*
--- nxlib/StrKeysym.c	2014-01-07 16:45:59.000000000 +0100
+++ nxlib.ok/StrKeysym.c	2014-02-26 11:13:14.000000000 +0100
@@ -276,7 +276,13 @@ XRefreshKeyboardMapping(XMappingEvent* e
 
 /* translate keycode to KeySym, no control/shift processing*/
 KeySym
-XKeycodeToKeysym(Display *dpy, unsigned int kc, int index)
+XKeycodeToKeysym(Display *dpy,
+#if NeedWidePrototypes
+    unsigned int kc       /* keycode */,
+#else
+    KeyCode kc            /* keycode */,
+#endif
+	int index)
 {
 	int	i;
 	MWKEY	mwkey;
diff -rupN nxlib/fontlist.c nxlib.ok/fontlist.c
--- nxlib/fontlist.c	2014-01-07 14:43:57.000000000 +0100
+++ nxlib.ok/fontlist.c	2014-01-07 14:42:21.000000000 +0100
@@ -24,13 +24,14 @@
  */
 char *FONT_DIR_LIST[] = {
 	"fonts",									/* local font dir w/fonts.alias*/
-	"/usr/share/fonts/X11/misc",				/* pcf fonts w/fonts.dir*/
-	"/usr/share/fonts/X11/100dpi",
-	"/usr/share/fonts/truetype/freefont",		/* truetype fonts, no fonts.dir*/
-	"/usr/share/fonts/truetype/ttf-dejavu",
-	"/usr/share/fonts/truetype/openoffice",
-	"/var/lib/defoma/x-ttcidfont-conf.d/dirs/TrueType",	/* truetype fonts, w/fonts.dir & fonts.scale*/
-	"/usr/share/fonts/X11/Type1",				/* t1lib type1 .pfb fonts, w/fonts.dir*/
+	"/usr/fonts",
+//	"/usr/share/fonts/X11/misc",				/* pcf fonts w/fonts.dir*/
+//	"/usr/share/fonts/X11/100dpi",
+//	"/usr/share/fonts/truetype/freefont",		/* truetype fonts, no fonts.dir*/
+//	"/usr/share/fonts/truetype/ttf-dejavu",
+//	"/usr/share/fonts/truetype/openoffice",
+//	"/var/lib/defoma/x-ttcidfont-conf.d/dirs/TrueType",	/* truetype fonts, w/fonts.dir & fonts.scale*/
+//	"/usr/share/fonts/X11/Type1",				/* t1lib type1 .pfb fonts, w/fonts.dir*/
 	0
 };
 
diff -rupN nxlib/Makefile nxlib.ok/Makefile
--- nxlib/Makefile	2014-01-07 14:43:57.000000000 +0100
+++ nxlib.ok/Makefile	2014-01-07 14:38:45.000000000 +0100
@@ -50,12 +50,13 @@ X11_RGBTXT=/usr/share/X11/rgb.txt
 # cross-compiling on a system with X11, no installation
 # is possible (as libX11.so would be replaced), and
 # the link command is changed from '-lX11' to '-lNX11 -lnano-X'
-LIBNAME=NX11
+#LIBNAME=NX11
+LIBNAME=X11
 xINSTALL_DIR=.
-INSTALL_DIR=/usr/local/lib
+INSTALL_DIR=$(ARM_SYSROOT)/usr/lib
 
 # set to Y to make shared libNX11.so library, shared lib dependencies
-SHAREDLIB=N
+SHAREDLIB=Y
 SOLIBREV=0.47
 SOEXTRALIBS = -L$(MWIN_LIB) -lnano-X
 
@@ -63,7 +64,7 @@ SOEXTRALIBS = -L$(MWIN_LIB) -lnano-X
 INCLUDE_XRM=Y
 
 # compiler flags
-CC = gcc
+CC = arm-linux-gcc
 LN = ln -s
 MV = mv
 CP = cp -p
diff -rupN nxlib/NextEvent.c nxlib.ok/NextEvent.c
--- nxlib/NextEvent.c	2014-01-07 14:43:57.000000000 +0100
+++ nxlib.ok/NextEvent.c	2014-01-07 14:38:45.000000000 +0100
@@ -210,7 +210,8 @@ translateNXEvent(Display *dpy, GR_EVENT 
 			event->xkey.y = pev->y;;
 			event->xkey.x_root = pev->rootx;
 			event->xkey.y_root = pev->rooty;
-			event->xkey.keycode = pev->scancode; /* note: not mwkey value*/
+			// CLM event->xkey.keycode = pev->scancode; /* note: not mwkey value*/
+			event->xkey.keycode = pev->ch; /* work with SDL but need rewrite sth. in fltk */
 			event->xkey.same_screen = True;
 
 			if (pev->modifiers & MWKMOD_CTRL)
diff -rupN nxlib/stub.c nxlib.ok/stub.c
--- nxlib/stub.c	2014-01-07 14:43:57.000000000 +0100
+++ nxlib.ok/stub.c	2014-01-07 14:38:45.000000000 +0100
@@ -160,3 +160,6 @@ int XRegisterIMInstantiateCallback() { D
 int XUnregisterIMInstantiateCallback() { DPRINTF("XUnregisterIMInstantiateCallback called\n"); return 0; } 
 int XIMOfIC() { DPRINTF("XIMOfIC called\n"); return 0; }
 int XLocaleOfIM() { DPRINTF("XLocaleOfIM called\n"); return 0; }
+
+int XChangeKeyboardControl(){ DPRINTF("XChangeKeyboardControl called\n");return 0;}
+int XOpenDevice() { DPRINTF("XOpenDevice called\n"); return 0; }
